vtm-playground/src/org/oscim/test/gdx/poi3d/GdxModelLayer.java

```diff
@@ -1,48 +1,88 @@
+/*
+ * Copyright 2014 Hannes Janetzek
+ * Copyright 2018 Gustl22
+ *
+ * This program is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
 package org.oscim.test.gdx.poi3d;
 
 import com.badlogic.gdx.assets.AssetManager;
 import com.badlogic.gdx.graphics.g3d.Model;
 import com.badlogic.gdx.graphics.g3d.ModelInstance;
 import com.badlogic.gdx.graphics.g3d.model.Node;
+import com.badlogic.gdx.utils.Array;
 
 import org.oscim.core.MapPosition;
+import org.oscim.core.MercatorProjection;
+import org.oscim.core.Tile;
 import org.oscim.event.Event;
+import org.oscim.gdx.GdxAssets;
 import org.oscim.layers.Layer;
+import org.oscim.layers.tile.buildings.BuildingLayer;
 import org.oscim.map.Map;
+import org.oscim.model.VtmModels;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.HashMap;
+
+/**
+ * Experimental layer to display 3d models.
+ */
 public class GdxModelLayer extends Layer implements Map.UpdateListener {
 
     static final Logger log = LoggerFactory.getLogger(GdxModelLayer.class);
 
-    GdxRenderer3D g3d;
+    public class ModelPosition {
```

Comment:
We can have `ModelPosition` in separate file (like `ModelHolder`) for more simplicity.

Reply:
Yes, like it's more comfortable :)